src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3238,6 +3238,27 @@ exprt c_typecheck_baset::do_special_functions(
                                       std::move(result),
                                       expr.source_location()};
   }
+  else if(
+    identifier == "__builtin_bitreverse8" ||
+    identifier == "__builtin_bitreverse16" ||
+    identifier == "__builtin_bitreverse32" ||
+    identifier == "__builtin_bitreverse64")
+  {
+    // clang only
+    if(expr.arguments().size() != 1)
+    {
+      error().source_location = f_op.source_location();
+      error() << identifier << " expects one operand" << eom;
+      throw 0;
```

Comment:
Throw a structured exception?
```suggestion
       std::ostringstream error_message;
      error_message << expr.source_location().as_string() << ": " << identifier
                    << " expects expects one operand";
      throw invalid_source_file_exceptiont{error_message.str()};
```

Reply:
Done, but I really wish we had #5837 (with its two dependencies) in place to make things even more structured!

Reply:
And now also includes a test for this one :-)